File: packages/core/lib/src/logging/console_logger.dart
import 'package:flutter/foundation.dart';

import 'logger.dart';

class ConsoleLogger implements Logger {
  @override
  void debug(String message) {
    if (kDebugMode) {
      print('[DEBUG] $message');
    }
  }

  @override
  void info(String message) {
    if (kDebugMode) {
      print('[INFO] $message');
    }
  }

  @override
  void warning(String message) {
    if (kDebugMode) {
      print('[WARNING] $message');
    }
  }

  @override
  void error(String message, {Object? error, StackTrace? stackTrace}) {
    if (kDebugMode) {
      print('[ERROR] $message');
      print(error);
      print(stackTrace);
    }
  }
}

File: backend/api/src/observability/console-logger.ts
import { Logger }
  from './logger';

export class ConsoleLogger
  implements Logger {

  info(
    message: string,
    metadata?: Record<string, unknown>,
  ): void {
    console.log(
      JSON.stringify({
        level: 'INFO',
        message,
        metadata,
      }),
    );
  }

  error(
    message: string,
    metadata?: Record<string, unknown>,
  ): void {
    console.error(
      JSON.stringify({
        level: 'ERROR',
        message,
        metadata,
      }),
    );
  }
}